packages/vm/src/static_analysis.rs

```diff
@@ -0,0 +1,204 @@
+use parity_wasm::elements::{deserialize_buffer, Internal, Module};
+use std::collections::HashSet;
+
+use crate::errors::{VmError, VmResult};
+
+pub const REQUIRED_IBC_EXPORTS: &[&str] = &[
+    "ibc_channel_open",
+    "ibc_channel_connect",
+    "ibc_channel_close",
+    "ibc_packet_receive",
+    "ibc_packet_ack",
+    "ibc_packet_timeout",
+];
+
+pub fn deserialize_wasm(wasm_code: &[u8]) -> VmResult<Module> {
+    deserialize_buffer(&wasm_code).map_err(|err| {
+        VmError::static_validation_err(format!(
+            "Wasm bytecode could not be deserialized. Deserialization error: \"{}\"",
+            err
+        ))
+    })
+}
+
+pub fn exported_functions(module: &Module) -> HashSet<String> {
```

Comment:
This was a bit confusing at first read - `parity_wasm::Module` (which is just parsed wasm) vs. `wasmer::Module` (which is JITed native code and a lot heavier). I don't see a way to improve, but nice to keep the parity_wasm type in this file and not in files that might refer to the `wasmer::Module`

Reply:
Yeah that was the idea: try keeping the different Modules in different parts of the code